File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution(object):
    def longestConsecutive(self, nums):
        mp = {}
        longest = 0
        for n in nums:
            if n not in mp:
                left = mp.get(n - 1, 0)
                right = mp.get(n + 1, 0)

                length = left + right + 1
                mp[n] = length 

                longest = max(longest, length)

                mp[n - left] = length
                mp[n + right] = length
        return longest
        """
        :type nums: List[int]
        :rtype: int
        """
             

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> count(26, 0);
        int left = 0, maxFreq = 0, maxLength = 0;

        for (int right = 0; right < s.length(); right++) {
            count[s[right] - 'A']++;
            maxFreq = max(maxFreq, count[s[right] - 'A']);

            int windowSize = right - left + 1;
            if (windowSize - maxFreq > k) {
                count[s[left] - 'A']--;
                left++;
            }

            maxLength = max(maxLength, right - left + 1);
        }
        return maxLength;
    }
};